SAMtags.tex

```diff
@@ -411,17 +411,111 @@ \section{Locally-defined tags}
 
 \begin{appendices}
 \appendix
-\section{SAM Tags History}\label{sec:history}
+\section{Tag History}
 
-This lists the date of each tagged SAM version along with changes that
-have been made while that version was current.  
+This appendix lists when standard tags were initially defined or significantly changed, and other historical events that affect how tags are interpreted or what files they may appear in.
 
-\subsection*{1.5: 23 May 2013 to current}
-\begin{itemize}
-\item Cellular barcode tags CB, CR, and CY added. (May 2018)
-\item Add UMI-related tags (RX, QX, OX, BZ, MI) and clarified usage of sample barcode tag BC. (August 2017)
-\item SAMtags.txt (this file) created with tags from SAMv1 
-\end{itemize}
+\setlength{\parindent}{0pt}
+\newcommand*{\gap}{\vspace*{2ex}}
+
+\subsubsection*{May 2018}
+
+Cellular barcode tags CB, CR, and CY added.
+
+\subsubsection*{November 2017}
+
+CG tag added.
+
+\subsubsection*{August 2017}
+
+Unique molecular identifier tags BZ, MI, OX, QX, and RX added.
+
+Usage of sample barcode tag BC clarified.
+
+\subsubsection*{June 2017}
+
+Corrected the description of the E2 tag, which previously wrongly claimed E2 to be encoded as per {\sf QUAL}.
+
+\subsubsection*{September 2016}
+
+Predefined tags, previously listed as a brief table within the main SAM specification, have been split out into this new document.
+There is now space for clearer and more complete tag descriptions.
+
+\subsubsection*{February 2014}
+
+MC tag added.
+
+\subsubsection*{May 2013}
+
+SAM version number {\tt VN:1.5} introduced, indicating the addition of the {\sc supplementary} flag (bit 0x800).
```

Comment:
I consider the SAM specification v1.5 milestone to also cover the de novo extensions in http://dx.doi.org/10.1101/020024 - can we add that here (with the URL)?

Reply:
That version number bump entry is there to mark (theoretical) compatibility concerns (see the VN:1.4 entry for a real example). Rather to my surprise, the CIGAR P operator was already in VN:1.0; are there other incompatibilities?

However I see your point and will reread that paper and fit something in. The URL would probably be even better in the main section (e.g. SAMv1.pdf's Guide for Describing Assembly Sequences in SAM) rather than this appendix.

Reply:
Our version control is a bit untidy.  CO and RT tags weren't there at the start of 1.4 but where there quite early into the 1.4 cycle and so were there at the end.  So while you could say 1.5 is the first release they were in from the start, we deem them to be a 1.4 feature.  Maybe it needs clarification for the CT/PT section though with a link in the "Describe padded alignments" comment (2011).

PS. Generally any behaviour that you *must* understand warrants a version update, rather than simply having the ability to ignore the new fields (tags mostly fall into this feature).  A new flag did (SUPPLEMENTARY), as did a complete new type of tag as the BAM parsing breaks ("B"), and belatedly as we'll update version for >= 65536 CIGAR entries.

Reply:
Thanks.

And yes, from memory the CIGAR P operator was already there, but this was may have been the first attempt to document how it could/should be used - does that sound about right @jkbonfield ?

Reply:
That description of how and when the version was bumped makes sense to me now, it seemed a lot more fluid / unclear to me back at the time.

Reply:
Yes, CIGAR P was there from very early on and there is even an example of it in the original samtools paper, complete with the expected output from tview.  Imagine my surprise when I discovered that samtools doesn't actually render the alignment in the paper as advertised!